### ISSUE TITLE
Semi-blockchain, phase 2

### ISSUE THREAD

**Issue by gmile** [edenlabllc/ops.api#16]
This is continuation of work started in https://github.com/edenlabllc/ops.api/pull/15. This issue will track semi-blockchain integration.

Main actionable steps are:

- [x] take verification function implemented as part of initial R&D effort, implement it as a cron job on `ops.api` service,
- [ ] audit verification function to make sure it makes use of DB indexes, and appropriate indexes are in place.

Answer the following questions:
- [x] <s>how often verification function is ran?</s> currently once a day
- [x] <s>every time verification function is ran, should it check entire chain or last day only?</s> entire chain is always preferable
- [ ] should terminated declarations be included in verification process? (Likely answer is _no, we should not exclude anything_).

  Consider the following scenario:
  1. declaration is created and signed,
  2. declaration remains active for the duration of `DECLARATION_TERM` (let's say 1 year),
  3. declaration is terminated.

  Now it's up to what verification strategy we've picking. If every time we're running verification we're re-building entire chain to compare each chain unit (A) to chain unit calculated once (B) in the past – we cannot throw away now-terminated declarations, since A and B won't match.

  In other words: A hash excludes terminated declaration, B hash was calculated when that declaration was active.

- [ ] depending on whether or not we're including terminated declaration, estimate the amount of declarations we'll have to verify after 1 year, after 5 years and after 15 years.
- [ ] estimate the time it takes to validate a chain of 1,000,000 declarations, see how the validation time grows with the amount of declaration growing linearly.

**Comment by lymychp** [comment]
all the tickets should be in [main repo](https://github.com/edenlabllc/ehealth.api/)
